File: task1.js
const str = process.argv[1];

function checkPalindrome(str){
    const strClean = str.replace(/\W/gi, '').toLowerCase();
    const strReverse = strClean.split("").reverse().join("");
    return strClean === strReverse ? "YES" : "NO";
}

process.stdout.write(checkPalindrome(str));

File: task1.rb
s = ''
for i in 0 ... ARGV.length
    s = s + ARGV[i]
end

def is_palindrome(s)
    s.downcase.gsub(/[\W_]/,'').eql?(s.downcase.gsub(/[\W_]/,'').reverse) ? 'YES' : 'NO'
end

if s.length > 0 then puts is_palindrome(s) end
